### ISSUE TITLE
Build fails when tappable is missing

### ISSUE THREAD

**Issue by TrevorSayre** [brunocodutra/webapp-webpack-plugin#139]
```
> project@1.0.0 build:common /Users/user/Documents/project
> cross-env NODE_ENV=production webpack --progress --profile --colors --mode production

  0% compiling/Users/user/Documents/project/node_modules/webapp-webpack-plugin/src/compiler.js:37
    compilation.hooks.webappWebpackPluginBeforeEmit = new AsyncSeriesWaterfallHook(['result']);
                                                      ^

TypeError: AsyncSeriesWaterfallHook is not a constructor
```

**Comment by brunocodutra** [comment]
Could you give more context on your setup? What versions of Webpack and webapp-webpack-plugin are you using? How are they configured?

**Comment by TrevorSayre** [comment]
```
"html-webpack-plugin": "^3.2.0",
"webpack": "^4.8.3",
"webapp-webpack-plugin": "^2.3.1",
```

I can include my entire `package.json` if it would help, but perhaps this is sufficient? I appreciate your quick response and am very thankful that you're addressing `npm audit` issues that `"favicons-webpack-plugin": "0.0.9",` has not.

**Comment by brunocodutra** [comment]
I attempted to reproduce the issue with the configuration below, but it seems to be working as expected.

#### package.json

```.js
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "build": "webpack -p"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "html-webpack-plugin": "3.2.0",
    "webapp-webpack-plugin": "2.3.1",
    "webpack": "4.8.3",
    "webpack-cli": "3.1.2"
  }
}
```

#### webpack.config.js

```.js
const HtmlWebpackPlugin = require('html-webpack-plugin');
const WebappWebpackPlugin = require('webapp-webpack-plugin')

module.exports = {

  entry: './src/entry.js',

  plugins: [
    new HtmlWebpackPlugin(),

    new WebappWebpackPlugin({
      logo: './logo.svg',
    }),
  ]
};

```

Could you share more details on your setup?

**Comment by dkrnl** [comment]
i catch similar bug:
this fix `npm i tapable@latest`

**Comment by GrumpyPirate** [comment]
We also ran into this issue when using `2.3.1`. It suddenly started happening on fresh installs. Versions: 
```
"webpack": "^4.25.1",
"webapp-webpack-plugin": "^2.3.1"
```

Downgrading to `"webapp-webpack-plugin": "2.2.0"` solved the issue. Upgrading `tapable` wasn't a workaround we were happy with. Any ideas what could be causing this issue?

**Comment by brunocodutra** [comment]
@GrumpyPirate to be honest I have no idea why that's happening to some of you and I haven't been able to reproduce it locally, but I suspect it has to do with the fact that `tapable` is not required as a direct dependency.

Would you be able to verify on your set-up, whether `"webapp-webpack-plugin": "brunocodutra/webapp-webpack-plugin#fix/139"` fixes the issue?

**Comment by temparus** [comment]
I had exactly the same issue. With your proposed fix in branch `fix/139` it works again.

**Comment by brunocodutra** [comment]
Thanks for testing @temparus!
The fix will be merged as part of #149.
